assets/src/apps/delivery/layouts/deck/components/HistoryNavigation.tsx

```diff
@@ -1,109 +1,121 @@
+    ?.reverse()
+    .map((entry) => entry.split('|')[0]);
 
-  // TODO this is actually driven by the student's history IF you are a student
-  // and the toc otherwise
-  // const historyItems =
-  //     currentPage.sequence
-  //         .filter((entry) => !entry.custom?.isLayer)
-  //         .map((entry) => {
-  //             return {
-  //                 id: entry.id,
-  //                 // TODO: pull all ensembles in sequence and get their names?
-  //                 // maybe need a history state to track this instead!
-  //                 name: entry.custom?.sequenceName || entry.id,
-  //             };
-  //         }) || [];
-  // const currentEnsembleIndex = historyItems.findIndex(
-  //     (item) => item.id === currentEnsemble.id
-  // );
-  // const isFirst = currentEnsembleIndex === 0;
-  // const isLast = currentEnsembleIndex === historyItems.length - 1;
+  // Get the activity names and ids to be displaeyd in histroy panel
+  const historyItems = globalSnapshot?.map((activityId) => {
+    const foundSequence = sequences.filter(
+      (sequence) => sequence.custom?.sequenceId === activityId,
+    )[0];
+    return {
+      id: foundSequence.custom?.sequenceId,
+      name: foundSequence.custom?.sequenceName || foundSequence.id,
+      timestamp: snapshot[`${foundSequence.custom?.sequenceId}|visitTimestamp`],
+    };
+  });
+  const currentEnsembleIndex = historyItems.findIndex((item: any) => item.id === currentActivityId);
+  const isFirst = currentEnsembleIndex === historyItems.length - 1;
+  const isLast = currentEnsembleIndex === 0;
+  const nextHandler = () => {
+    const prevActivity = historyItems[currentEnsembleIndex - 1];
+    dispatch(navigateToActivity(prevActivity.id));
+  };
 
+  const prevHandler = () => {
+    const prevActivity = historyItems[currentEnsembleIndex + 1];
+    dispatch(navigateToActivity(prevActivity.id));
+  };
   return (
     <Fragment>
-      {/* {enableHistory ? (
-                <div className="historyStepView pullLeftInCheckContainer">
-                    <div className="historyStepContainer">
-                        <button
-                            onClick={prevHandler}
-                            className="backBtn historyStepButton"
-                            aria-label="Previous screen"
-                            disabled={isFirst}
-                        >
-                            <span className="icon-chevron-left" />
-                        </button>
-                        <button
-                            onClick={nextHandler}
-                            className="nextBtn historyStepButton"
-                            aria-label="Next screen"
-                            disabled={isLast}
-                        >
-                            <span className="icon-chevron-right" />
-                        </button>
-                    </div>
-                </div>
-            ) : null} */}
+      {isHistoryModeOn && !previewMode ? (
```

Comment:
why not in preview mode? in preview they can still do this, they just should see it exactly as the student would